modules/rntuple.mjs

```diff
@@ -747,35 +732,53 @@ class RNTupleDescriptorBuilder {
+// Example Of Deserializing Page Content
+deserializePage(blob, columnDescriptor, pageInfo) {
+    const originalColtype = columnDescriptor.coltype,
+    { coltype } = recontructUnsplitBuffer(blob, columnDescriptor);
+    let { blob: processedBlob } = recontructUnsplitBuffer(blob, columnDescriptor);
 
-        
-        // Handle split index types
+    
+    // Handle split index types
         if (originalColtype === ENTupleColumnType.kSplitIndex32 || originalColtype=== ENTupleColumnType.kSplitIndex64) {
-            const { blob: decodedArray } = DecodeDeltaIndex(processedBlob, coltype);
-            processedBlob = decodedArray;  
-        }
+        const { blob: decodedArray } = DecodeDeltaIndex(processedBlob, coltype);
+        processedBlob = decodedArray;  
+    }
 
-        // Handle Split Signed Int types
-        if (originalColtype === ENTupleColumnType.kSplitInt16 || originalColtype === ENTupleColumnType.kSplitInt32 || originalColtype === ENTupleColumnType.kSplitInt64) {
-            const { blob: decodedArray } = decodeZigzag(processedBlob, coltype);
-            processedBlob = decodedArray;  
-        }
+    // Handle Split Signed Int types
+    if (originalColtype === ENTupleColumnType.kSplitInt16 || originalColtype === ENTupleColumnType.kSplitInt32 || originalColtype === ENTupleColumnType.kSplitInt64) {
+        const { blob: decodedArray } = decodeZigzag(processedBlob, coltype);
+        processedBlob = decodedArray;  
+    }
 
-        const byteSize = getTypeByteSize(coltype),
-              reader = new RBufferReader(processedBlob),
-              values = [];
+    const reader = new RBufferReader(processedBlob),
+          values = [],
 
-        if (!byteSize)
-            throw new Error('Invalid or unsupported column type: cannot determine byte size');
+    // Use numElements from pageInfo parameter
+    numValues = Number(pageInfo.numElements);
 
-        const numValues = processedBlob.byteLength / byteSize;
+    switch (coltype) {
+        case ENTupleColumnType.kBit: {
+            let bitCount = 0;
+            const totalBitsInBuffer = processedBlob.byteLength * 8;
+            
+            for (let byteIndex = 0; byteIndex < processedBlob.byteLength; ++byteIndex) {
+                const byte = reader.readU8();
+                
+                // Extract 8 bits from this byte
+                for (let bitPos = 0; bitPos < 8 && bitCount < totalBitsInBuffer && bitCount < numValues; ++bitPos, ++bitCount) {
+                    const bitValue = (byte >>> bitPos) & 1,
+                    boolValue = bitValue === 1;
+                    values.push(boolValue);
+                }
+                
+                if (bitCount >= numValues) break;
+            }
+            break;
+        }
 
-        for (let i = 0; i < (numValues ?? processedBlob.byteLength); ++i) {
+        default: {
+        for (let i = 0; i < numValues; ++i) {
             let val;
             
             switch (coltype) {
```

Comment:
Why have a `switch (coltype)` nested within another `switch (coltype)`? You should extract the inner cases into the outer switch (and have the `for` be inside each case).
To avoid repeating the same `for` over and over again I suggest extracting it into a function (maybe one local to this function) that does the job. Something like:

```js
let extractValues = (readFunc) => {
  for (let i = 0; i < numValues; ++i)
    values.push(readFunc());
}

switch (coltype) {
case ENTupleColumnType.kBit:
  // ... handle Bit ...
case ENTupleColumnType.kReal64:
   extractValues(reader.readF64.bind(reader));
   break;
case ENTupleColumnType.kReal32:
   extractValues(reader.readF32.bind(reader));
   break;
// ...
}
```